packages/schemas/src/types/hook.ts

```diff
@@ -13,3 +21,21 @@ export type HookEventPayload = {
   user?: Pick<User, (typeof userInfoSelectFields)[number]>;
   application?: Pick<Application, 'id' | 'type' | 'name' | 'description'>;
 } & Record<string, unknown>;
+
+const nonemptyUniqueHookEventsGuard = hookEventsGuard
+  .nonempty()
+  .transform((events) => deduplicate(events));
+
+export const createHookGuard = object({
+  // Note: ensure the user will not create a hook with an empty name.
+  name: string().min(1).optional(),
+  event: hookEventGuard.optional(),
+  events: nonemptyUniqueHookEventsGuard.optional(),
+  config: hookConfigGuard,
+  enabled: boolean().optional(),
+});
+
+export const updateHookGuard = createHookGuard
+  .omit({ events: true })
+  .deepPartial()
```

Comment:
how will `deepPartial()` affect `events`? even it comes with unexpected outcome will `.partial().extend({ config: ... })` be more clear?

Reply:
This is an error implementation, already updated by following our API convention, and we will update the whole config and event through the `PATCH /hooks/:id` API.